都 2021 年了，你还不知道这些检查 CSS 的工具？ (#8432)
xitu/gold-miner: article/2021/tools-for-auditing-css.md

> * 原文作者：[Silvestar Bistrović](https://css-tricks.com/author/silvestar/)
> * 译文出自：[掘金翻译计划](https://github.com/xitu/gold-miner)
> * 本文永久链接：[https://github.com/xitu/gold-miner/blob/master/article/2021/tools-for-auditing-css.md](https://github.com/xitu/gold-miner/blob/master/article/2021/tools-for-auditing-css.md)
> * 译者：[霜羽 Hoarfroster](https://github.com/Hoarfroster)

> > * 校对者：

Author: ```suggestion
> * 校对者：[zenblo](https://github.com/zenblo)
```

# 不会吧不会吧，都 1202 年了，你还不知道这些检查 CSS 的工具？？？

> 检查 CSS 在开发者的日常生活中并不是一项常见的任务，但有时我们却不得不去做。也许这是性能检查的一部分，以确定有问题的 CSS 或删去未使用的选择器。也许是努力提高可访问性的一部分，在代码库中使用的所有颜色都要进行对比评估。甚至可能是为了加强一致性。

Reviewer: ```suggestion
在开发者的日常工作中检查 CSS 并不是一项常见的任务，但有时却不得不做。或许这是性能检查的一部分，用来排查有问题的 CSS 或删去未使用的选择器。也许是努力提高可访问性的一部分，在代码库中使用的所有颜色都要进行对比评估。甚至可能是为了加强一致性。
```

不管是什么情况，每当需要检查 CSS 的时候到来，我通常都会使用我将在下文中向大家介绍的那些工具。但在此之前，让我们先看看检查 CSS 到底是什么。

> # 目录

Reviewer: ```suggestion
## 目录
```
二级目录这些不需要修改，会影响排版

Reviewer 2: 不管什么情况，每当需要检查CSS的时候，我通常会使用下文中向大家介绍的一些工具。在此之前，让我们先看看检查CSS到底是什么。

1. 浏览器内的开发工具
2. 在线工具
3. CLI 工具

> # 检查 CSS 是很难的

Reviewer: ```suggestion
## 检查 CSS 有难度
```

> 一般来说，代码检查涉及分析代码，以发现 bug 或其他不对劲的地方，就比如说那些可能存在的性能上的问题。对于大多数编程语言来说，检查代码的概念相对简单：它正常工作了还是没能够正常工作。但 CSS 是一种特殊的语言，错误大多都会被浏览器忽略。而且我们能够[用很多不同的方式实现相同的样式](https://css-tricks.com/hearts-in-html-and-css/)，这就使得 CSS 的检查变得有些棘手 —— 至少可以这么说。

Reviewer: ```suggestion
一般来说，代码检查涉及分析代码，排除 bug 或其他不恰当的地方，就比如说那些可能存在的性能问题。对于大多数编程语言来说，检查代码的概念相对简单：它正常工作了或者没能够正常工作。但 CSS 是一种特殊的语言，错误大多都会被浏览器忽略。而且我们能够[用很多不同的方式实现相同的样式](https://css-tricks.com/hearts-in-html-and-css/)，这就使得 CSS 的检查变得有些棘手 —— 至少可以这么说。
```

通过使用你最喜欢的代码编辑器的扩展程序，或者设置一个 CSS 的 linter 或 CSS 代码检查器可能会帮助到你及时发现这些错误，不过这不是我想在这里展示的，而且我们不应该止步于此。因为我们可以使用[太多](https://css-tricks.com/a-quick-css-audit-and-general-notes-about-design-systems/)的颜色、排版定义或 `z-index` 值，而所有这些都可能让一个 CSS 代码库变得混乱、不可维护且不稳定。

要真正检查 CSS，我们需要深入挖掘并找到那些不被认为是最佳实践的地方。为了找到这些地方，我们可以使用以下工具：

> # 浏览器内的开发工具

Reviewer: ```suggestion
## 浏览器内置的开发工具
```
